forest/willow/log_stats.py

```diff
@@ -1,160 +1,373 @@
+                num_out_call,
+                num_mis_call,
+                num_uniq_in_call,
+                num_uniq_out_call,
+                num_uniq_mis_call,
+                total_time_in_call,
+                total_time_out_call,
+                num_s,
+                num_r,
+                num_mms_s,
+                num_mms_r,
+                num_s_tel,
+                num_r_tel,
+                total_char_s,
+                total_char_r,
+            ]
         summary_stats.append(newline)
-    if option == 'daily':
-        stats_pdframe = pd.DataFrame(summary_stats, columns=['year', 'month', 'day','num_in_call', 'num_out_call', 'num_mis_call',
-                'num_in_caller', 'num_out_caller','num_mis_caller', 'total_mins_in_call', 'total_mins_out_call',
-                'num_s', 'num_r', 'num_mms_s', 'num_mms_r', 'num_s_tel','num_r_tel', 'total_char_s', 'total_char_r',
-                'text_reciprocity_incoming','text_reciprocity_outgoing'])
-    if option == 'hourly':
-        stats_pdframe = pd.DataFrame(summary_stats, columns=['year', 'month', 'day','hour','num_in_call', 'num_out_call',
-                'num_mis_call','num_in_caller', 'num_out_caller','num_mis_caller', 'total_mins_in_call', 'total_mins_out_call',
-                'num_s', 'num_r', 'num_mms_s', 'num_mms_r', 'num_s_tel','num_r_tel', 'total_char_s', 'total_char_r'])
+    if frequency == Frequency.DAILY:
+        stats_pdframe = pd.DataFrame(
+            summary_stats,
+            columns=[
+                "year",
+                "month",
+                "day",
+                "num_in_call",
+                "num_out_call",
+                "num_mis_call",
+                "num_in_caller",
+                "num_out_caller",
+                "num_mis_caller",
+                "total_mins_in_call",
+                "total_mins_out_call",
+                "num_s",
+                "num_r",
+                "num_mms_s",
+                "num_mms_r",
+                "num_s_tel",
+                "num_r_tel",
+                "total_char_s",
+                "total_char_r",
+                "text_reciprocity_incoming",
+                "text_reciprocity_outgoing",
+            ],
+        )
+    else:
+        stats_pdframe = pd.DataFrame(
+            summary_stats,
+            columns=[
+                "year",
+                "month",
+                "day",
+                "hour",
+                "num_in_call",
+                "num_out_call",
+                "num_mis_call",
+                "num_in_caller",
+                "num_out_caller",
+                "num_mis_caller",
+                "total_mins_in_call",
+                "total_mins_out_call",
+                "num_s",
+                "num_r",
+                "num_mms_s",
+                "num_mms_r",
+                "num_s_tel",
+                "num_r_tel",
+                "total_char_s",
+                "total_char_r",
+            ],
+        )
```

Comment:
Simplify `columns` creation the same way as `newline` above? Then you can also get rid `stats_pdframe` variable:
```
return pd.DataFrame(summary_stats, columns)
```